playbooks/allocate_pubcloud.yml

```diff
@@ -59,13 +64,41 @@
             timeout: "{{ boot_timeout }}"
           register: _instance_provision
 
+        - name: Write inventory
+          copy:
+            content: |
+              [job_nodes:children]
+              hosts
+
+              [hosts]
+              {{ _instance_provision.server.name }} ansible_host={{ _instance_provision.server.accessIPv4 }} ansible_user=root rax_region={{ _instance_provision.server.region }}
+            dest: '{{ inventory_path }}/hosts'
+
         - name: Wait for SSH connectivity to the cloud instance (10 min timeout)
-          wait_for_connection:
+          wait_for:
             timeout: 600
             port: 22
             host: "{{ _instance_provision.server.accessIPv4 }}"
+            sleep: 10
+            search_regex: "OpenSSH"
 
       rescue:
+        - name: Ensure first attempt instance is removed.
+          os_server:
+            name: "{{ instance_name }}"
+            flavor: "{{ flavor }}"
+            state: absent
+            cloud: "{{ cloud_name }}"
+            region_name: "{{ regions_shuff[0] }}"
+            image: "{{ image }}"
+            key_name: "{{ keyname }}"
```

Comment:
You can probably get rid of a lot of these pairs because they're only needed when creating the instance.

Reply:
Maybe call the clean up playbook here instead of having a task?